### PR TITLE
Feature 172 documentation

### LINKED ISSUE #172: Provide missing documentation to the Users Guide
## Describe the Task ##
Some newly added plots do not have user documentation.  

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [x] Wind rose
- [x] Equivalence-bounds testing (example in METviewer https://metviewer.readthedocs.io/en/develop/Users_Guide/etb_plot.html) <- done by Tatiana for an earlier PR
- [x] Difficulty index (remember to create a link to the METplus Use Case )
- [x]  Probability, Relative, Rank Histogram
- [x] ECLV (documentation in METviewer User's Guide)
- [x] Hovmoeller plot needs update to the instructions for running from the command line 

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
Split 2792541, 2793541

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by bikegeek**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
    visually inspected the documentation created on [readthedocs.io](https://metplotpy.readthedocs.io/en/feature_172_documentation/Users_Guide/index.html) to verify that the following changes were made:

    - for all plots, the instructions to set the METPLOTPY_BASE is now set to _$METPLOTPY_SOURCE/METplotpy_
 (previously set to _$METPLOTPY_SOURCE/METplotpy/metplotpy_ but changes were made in base_plot.py requiring this change).

    - Documentation now exists for:
        -  Difficulty Index
        -  Hovmoeller index is correct and now has the following instructions for running on the command line: 
        _python $METPLOTPY_SOURCE/METplotpy/metplotpy/plots/hovmoeller/hovmoeller.py --config $WORKING_DIR/hovmoeller_defaults.yaml --datadir $WORKING_DIR/data/hovmoeller  --input precip.erai.sfc.1p0.2x.2014-2016.nc_
        - Economic Cost/Loss Value (ECLV) plot
        - Wind Rose

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
   - Verify that the documentation exists for the:
       - Wind Rose
       - Difficulty Index
       - Hovmoeller 
       - ECLV (Economic Cost/Loss Value)
       - Histogram (rank, probability, and relative frequency)
  - Verify for all plots that the METPLOTPY_BASE is to be set to _$METPLOTPY_SOURCE/METplotpy_

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**
     - fixed and updated the tests for the histograms

- [x] Will this PR result in changes to the test suite? **[Yes ]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
   - the histogram tests for rank, probability, and relative frequency have been fixed and
      added to the list of tests to run in Github actions

- [ ] Please complete this pull request review by **Thursday March 3**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by jprestop** [comment]
@bikegeek 
Hi.  I'm just starting on this now.  Hera is down for maintenance so I am starting on Orion.  In this PR it says:

> for all plots, the instructions to set the METPLOTPY_BASE is now set to $METPLOTPY_SOURCE/METplotpy
> (previously set to $METPLOTPY_SOURCE/METplotpy/metplotpy but changes were made in base_plot.py requiring this change).

I just grabbed beta6 and unpacked it.  However, when I ran the following:
```
Orion-devel-1[22] jprestop$ ls /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/
```
I get an error:
```
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/: No such file or directory
```

Should the final "METplotpy" in the path instead be "metplotpy", which does exist?
```
Orion-devel-1[22] jprestop$ ls /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/
contributed  __init__.py  LICENSE  plots  README  README.md
```

**Comment by jprestop** [comment]
@bikegeek 
I was going to proceed with presuming that the things in my previous comment were true.  I started following along in this section to try to run the line plot:
https://metplotpy.readthedocs.io/en/feature_172_documentation/Users_Guide/line.html#custom-configuration-file

Unfortunately, following this direction did not work:

> Modify the stat_input setting in the $METPLOTPY_SOURCE/METplotpy/test/line/custom_line.yaml file to explicitly point to the $METPLOTPY_SOURCE/METplotpy/test/line directory (where the custom config files and sample data reside). 

I started by running 'ls' on the first path listed and modified it until it worked:
```
Orion-devel-1[46] jprestop$ ls $METPLOTPY_SOURCE/METplotpy/test/line/custom_line.yaml
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/test/line/custom_line.yaml: No such file or directory
Orion-devel-1[47] jprestop$ ls $METPLOTPY_SOURCE/metplotpy/test/line/custom_line.yaml
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/test/line/custom_line.yaml: No such file or directory
Orion-devel-1[48] jprestop$ ls $METPLOTPY_SOURCE/test/line/custom_line.yaml
/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/test/line/custom_line.yaml
```

It looks like there are areas of the documentation that need to be fixed.  I'm sorry to be the bearer of bad news.  Please let me know when you are ready for me to re-review, and I'll start with the line plot again.

File: docs/Users_Guide/line.rst [hunk 1]
```diff
@@ -17,14 +17,22 @@ Example
...
 ___________
 
-The sample data used to create an example line plot is available in the
-*METplotpy* repository, where the line plot tests are located:
+The data is text output from MET in columnar format. The sample data used to
+create an example line plot is available in the
+`METplotpy <https://github.com/dtcenter/METplotpy>`_ repository, where the line plot tests are located:
 
-*$METPLOTPY_SOURCE/METplotpy/metplotpy/test/line/line.data*
+*$METPLOTPY_BASE/metplotpy/test/line/line.data*
```
Comment (jprestop) [inline docs/Users_Guide/line.rst:24]:
```suggestion
*$METPLOTPY_BASE/test/line/line.data*
```
I believe "metplotpy" needs to be removed according to:
```
Orion-devel-1[60] jprestop$ ls $METPLOTPY_BASE/metplotpy/test/line/line.data
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6//metplotpy/test/line/line.data: No such file or directory
Orion-devel-1[62] jprestop$ echo $METPLOTPY_BASE/test/line/line.data
/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6//test/line/line.data
```

Comment (jprestop) [inline docs/Users_Guide/line.rst:24]:
```suggestion
*$METPLOTPY_BASE/test/line/line.data*
```
@bikegeek I believe "metplotpy" needs to be removed according to:
```
Orion-devel-1[60] jprestop$ ls $METPLOTPY_BASE/metplotpy/test/line/line.data
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6//metplotpy/test/line/line.data: No such file or directory
Orion-devel-1[62] jprestop$ echo $METPLOTPY_BASE/test/line/line.data
/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6//test/line/line.data
```

File: docs/Users_Guide/line.rst [hunk 2]
```diff
@@ -103,13 +113,13 @@ For example:
...
 *$WORKING_DIR* directory that is specified exists and has the appropriate
 read and write permissions.  The path listed for *plot_filename* may be
-changed to the output directory of one's  choosing.  If this is not set,
+changed to the output directory of one's choosing.  If this is not set,
 then the *plot_filename* setting specified in the
-*$METPLOTPY_SOURCE/METplotpy/metplotpy/plots/config/line_defaults.yaml*
+*$METPLOTPY_BASE/metplotpy/plots/config/line_defaults.yaml*
 configuration file will be used.
 
 To save the intermediate **.points1** file (used by METviewer and is useful
```
Comment (jprestop) [inline docs/Users_Guide/line.rst:125]: @bikegeek, I do not see a points_path in the $METPLOTPY_BASE/test/line/custom_line.yaml file.

**Comment by jprestop** [comment]
@bikegeek Thanks for updating the paths and the directions in the line.rst file.  I walked through them and made suggestions to change one path, as the path listed did not work for me.  I also noted in a comment that the points_path does not exist in custom_line.yaml.  Once I added points_path and a value for it to my custom_line.yaml file, I was able to run the following successfully on Orion:
```
python /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/plots/line/line.py ./custom_line.yaml
```
Please let me know once you have updated the other files and I will start to review the other plots.  Thank you for all of your work on this task!

**Comment by jprestop** [comment]
Hi @bikegeek.  Below are a few changes that need to be made to correct links in performance_diagram.rst.  After typing them up, I realized that they all have to do with the need for "$METPLOTPY_SOURCE/METplotpy/metplotpy" to be "$METPLOTPY_SOURCE/metplotpy" or "$METPLOTPY_SOURCE/METplotpy/test/" to be "$METPLOTPY_SOURCE/test/".  A quick check shows that this exists in other .rst files as well.  Hopefully, it will be a quick change to remove "METplotpy" from the paths.  Please let me know when you are ready for me to re-review.  Thanks!

Under "Sample Data", I believe "$METPLOTPY_SOURCE/METplotpy/test/performance_diagram/plot_20200317_151252.data" should be "$METPLOTPY_SOURCE/test/performance_diagram/plot_20200317_151252.data":

```
Orion-devel-1[17] jprestop$ ls $METPLOTPY_SOURCE/METplotpy/test/performance_diagram/plot_20200317_151252.data
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/test/performance_diagram/plot_20200317_151252.data: No such file or directory
Orion-devel-1[18] jprestop$ ls $METPLOTPY_SOURCE/test/performance_diagram/plot_20200317_151252.data
/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/test/performance_diagram/plot_20200317_151252.data
```

Under "Configuration File", I believe "$METPLOTPY_SOURCE/METplotpy/metplotpy/plots/config" should be "$METPLOTPY_SOURCE/metplotpy/plots/config" in two places:
```
Orion-devel-1[19] jprestop$ ls $METPLOTPY_SOURCE/METplotpy/metplotpy/plots/config
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/metplotpy/plots/config: No such file or directory
Orion-devel-1[20] jprestop$ ls $METPLOTPY_SOURCE/metplotpy/plots/config
bar_defaults.yaml      ens_ss_defaults.yaml			 histogram_defaults.yml    performance_diagram_defaults.yaml  taylor_diagram_defaults.yaml
box_defaults.yaml      equivalence_testing_bounds_defaults.yaml  hovmoeller_defaults.yaml  reliability_defaults.yaml	      wind_rose_defaults.yaml
contour_defaults.yaml  hist_defaults.yaml			 line_defaults.yaml	   roc_diagram_defaults.yaml
eclv_defaults.yaml     histogram_2d_defaults.yaml		 mpr_plot_defaults.yaml    scatter_defaults.yaml
```

Under "Custom Configuration File", I believe "cp $METPLOTPY_SOURCE/METplotpy/test/performance_diagram/custom_performance_diagram.yaml $WORKING_DIR/custom_performance_diagram.yaml" should be "cp $METPLOTPY_SOURCE/test/performance_diagram/custom_performance_diagram.yaml $WORKING_DIR/custom_performance_diagram.yaml":
```
Orion-devel-1[21] jprestop$ cp $METPLOTPY_SOURCE/METplotpy/test/performance_diagram/custom_performance_diagram.yaml .
cp: cannot stat ‘/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/test/performance_diagram/custom_performance_diagram.yaml’: No such file or directory
Orion-devel-1[22] jprestop$ cp $METPLOTPY_SOURCE/test/performance_diagram/custom_performance_diagram.yaml .
```
There are also other (at least three) references to "$METPLOTPY_SOURCE/METplotpy/test/performance_diagram/custom_performance_diagram.yaml" in that same section that should be "$METPLOTPY_SOURCE/test/performance_diagram/custom_performance_diagram.yaml".

Also in that section "$METPLOTPY_SOURCE/METplotpy/test/performance_diagram/plot_20200317_151252.data" needs to be "$METPLOTPY_SOURCE/test/performance_diagram/plot_20200317_151252.data":
```
Orion-devel-1[24] jprestop$ ls $METPLOTPY_SOURCE/METplotpy/test/performance_diagram/plot_20200317_151252.data
ls: cannot access /apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/METplotpy/test/performance_diagram/plot_20200317_151252.data: No such file or directory
Orion-devel-1[25] jprestop$ ls $METPLOTPY_SOURCE/test/performance_diagram/plot_20200317_151252.data
/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/test/performance_diagram/plot_20200317_151252.data
```

Under "Using Defaults", I believe "cp $METPLOTPY_SOURCE/METplotpy/test/performance_diagram/minimal_performance_diagram.yaml $WORKING_DIR/minimal_performance_diagram.yaml" needs to be "cp $METPLOTPY_SOURCE/test/performance_diagram/minimal_performance_diagram.yaml $WORKING_DIR/minimal_performance_diagram.yaml".

And, there are two references to "$METPLOTPY_SOURCE/METplotpy/test/performance_diagram/plot_20200317_151252.data" which should be "$METPLOTPY_SOURCE/test/performance_diagram/plot_20200317_151252.data".

Under "Run from the Command Line":
There are three references to "$METPLOTPY_SOURCE/METplotpy/metplotpy" that should be "$METPLOTPY_SOURCE/metplotpy". 

"python $METPLOTPY_SOURCE/METplotpy/metplotpy/plots/performance_diagram/performance_diagram.py $WORKING_DIR/minimal_performance_diagram.yaml" should be "python $METPLOTPY_SOURCE/metplotpy/plots/performance_diagram/performance_diagram.py $WORKING_DIR/minimal_performance_diagram.yaml" 

"python $METPLOTPY_SOURCE/METplotpy/metplotpy/plots/performance_diagram/performance_diagram.py $WORKING_DIR/custom_performance_diagram.yaml" should be "python $METPLOTPY_SOURCE/metplotpy/plots/performance_diagram/performance_diagram.py $WORKING_DIR/custom_performance_diagram.yaml"

**Comment by jprestop** [comment]
@bikegeek 
I decided to go ahead and continue testing the Performance Diagram, knowing what the paths should be.  Unfortunately, I got an error when running:
```
python ${METPLOTPY_SOURCE}/metplotpy/plots/performance_diagram/performance_diagram.py ./custom_performance_diagram.yaml
```
```
Orion-devel-1[35] jprestop$ python ${METPLOTPY_SOURCE}/metplotpy/plots/performance_diagram/performance_diagram.py ./custom_performance_diagram.yaml
Traceback (most recent call last):
  File "/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/plots/performance_diagram/performance_diagram.py", line 17, in <module>
    from .performance_diagram_config import PerformanceDiagramConfig
ImportError: attempted relative import with no known parent package
```

While the line plot worked for me, the performance diagram is not working for me due to the error above.

**Comment by jprestop** [comment]
@bikegeek Good news and bad news.  Good news - for my long post from earlier today, I was looking in the develop branch and not this feature branch.  (So sorry!).  Bad news - There are still references in the feature branch to $METPLOTPY_SOURCE/METplotpy that should be $METPLOTPY_SOURCE/metplotpy.  So, it does still need some work, but hopefully not as much.

**Comment by jprestop** [comment]
@bikegeek It looks like points_path is missing from custom_eclv.yaml.

**Comment by jprestop** [comment]
@bikegeek, it looks like something has run amok with the Table of Contents in the feature branch.  I will attach two screen shots, one from develop and one from this feature branch.  "develop" shows line, box, bar, ensemble spread-skill, contour, etc.  The feature branch shows line, box, bar, eclv, but then it shows **description, example METplus configuration, run from the command line** (which are all subsections, I believe from ensemble spread-skill), contour.

**Comment by jprestop** [comment]
@bikegeek It looks like points_path is missing from custom_contour.yaml.

**Comment by jprestop** [comment]
@bikegeek And, with regard to my points_path missing comments, I am going off of beta6, but I see that you already caught those here.  I will check first before posting next time.

**Comment by jprestop** [comment]
@bikegeek In testing the Difficulty Index, there are references to $METPLOTPY_SOURCE/METplotpy.  Ignoring those and using the correct path, the directions say:
```
  Run the following on the command line:
  python example_difficulty_index.py
```

running that line from my working directory doesn’t work:
```
Orion-devel-1[73] jprestop$ python example_difficulty_index.py
python: can't open file 'example_difficulty_index.py': [Errno 2] No such file or directory
```
But running the following does:
```
Orion-devel-1[76] jprestop$ python   ${METPLOTPY_SOURCE}/test/difficulty_index/example_difficulty_index.py
Saving ./swh_North_Pacific_difficulty_index_9_00_feet.png...

Saving ./swh_North_Pacific_difficulty_index_10_00_feet.png...

Saving ./swh_North_Pacific_difficulty_index_11_00_feet.png...

Saving ./swh_North_Pacific_difficulty_index_12_00_feet.png...

Saving ./swh_North_Pacific_5dy_mean.png...

Saving ./swh_North_Pacific_5dy_std.png...
```

However, it does not allow the user to chose where they want the output to go because there is no config file, which you likely already know.  Would it be possible to add a configuration file or a command line option to allow the user to decide where to put the output?  I think generally, we'd like to have that as a user configurable option.

**Comment by jprestop** [comment]
@bikegeek In testing the ROC Diagram, there are references to $METPLOTPY_SOURCE/METplotpy.  Ignoring those and running:
```
python  ${METPLOTPY_SOURCE}/metplotpy/plots/roc_diagram/roc_diagram.py ./custom_roc_diagram.yaml
```
I get the following error:
```
Orion-devel-1[87] jprestop$ python  ${METPLOTPY_SOURCE}/metplotpy/plots/roc_diagram/roc_diagram.py ./custom_roc_diagram.yaml
while parsing a block mapping
  in "./custom_roc_diagram.yaml", line 4, column 1
expected <block end>, but found '<block mapping start>'
  in "./custom_roc_diagram.yaml", line 10, column 2
Traceback (most recent call last):
  File "/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/plots/roc_diagram/roc_diagram.py", line 470, in <module>
    main()
  File "/apps/contrib/MET/METplotpy/METplotpy-1.1.0-beta6/metplotpy/plots/roc_diagram/roc_diagram.py", line 459, in main
    r = ROCDiagram(docs)
UnboundLocalError: local variable 'docs' referenced before assignment
```

File: docs/Users_Guide/difficulty_index.rst [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      cd $WORKING_DIR
+
+where *$WORKING_DIR* is the directory where you copied all the necessary files (e.g. /home/users/someuser/working_dir).
+
+
+* Run the following on the command line:
+
+  .. code-block:: ini
+
+    python example_difficulty_index.py example_difficulty_index.yaml
```
Comment (jprestop) [inline docs/Users_Guide/difficulty_index.rst]:
```suggestion
    python ${METPLOTPY_BASE}/test/difficulty_index/example_difficulty_index.py --config $WORKING_DIR/example_difficulty_index.yaml
```
The command that is currently listed did not work for me, but the one I am suggesting did.

File: docs/Users_Guide/polar_plot.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+*model_applications/marine_coastal/GridStat_fcstRTOFS_obsOSTIA_iceCover/grid_stat_north_000000L_20210305_120000V_pairs.nc.*
+
+Save this file in a directory where you have read and write permissions, such as
+$WORKING_DIR/data/polar_plot, where $WORKING_DIR is the path to your data directory.
+
+Configuration Files
+___________________
+
+There is a YAML config file located in
+*$METPLOTPY_BASE/metplotpy/plots/config/polar_plot_defaults.yaml*
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]: I do not see this file in the repo:
```
Orion-dtn-4[130] jprestop$ ls /apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation/metplotpy/plots/config
bar_defaults.yaml      equivalence_testing_bounds_defaults.yaml  line_defaults.yaml		    scatter_defaults.yaml
box_defaults.yaml      hist_defaults.yaml			 mpr_plot_defaults.yaml		    taylor_diagram_defaults.yaml
contour_defaults.yaml  histogram_2d_defaults.yaml		 performance_diagram_defaults.yaml  wind_rose_defaults.yaml
eclv_defaults.yaml     histogram_defaults.yml			 reliability_defaults.yaml
ens_ss_defaults.yaml   hovmoeller_defaults.yaml			 roc_diagram_defaults.yaml
```

File: docs/Users_Guide/polar_plot.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+Copy this configuration file from where you saved the METplotpy source code to your working directory:
+
+.. code-block:: ini
+
+  cp $METPLOTPY_BASE/metplotpy/plots/config/polar_plot.yaml $WORKING_DIR/polar_plot.yaml
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
  cp $METPLOTPY_BASE/metplotpy/plots/polar_plot/polar_ice.yaml $WORKING_DIR/polar_ice.yaml
```

File: docs/Users_Guide/polar_plot.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+$METPLOTPY_BASE is the directory where you saved the METplotpy code, and $WORKING_DIR is the directory where you
+have read and write permissions.
+
+
+Run from the Command Line
+=========================
+
+To generate the example Polar Plot plot (i.e. using settings in the
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
To generate the example Polar Ice plot (i.e. using settings in the
```

File: docs/Users_Guide/polar_plot.rst [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+**polar_plot.yaml** configuration file) perform the following:
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
**polar_ice.yaml** configuration file) perform the following:
```

File: docs/Users_Guide/polar_plot.rst [hunk 8]
```diff
@@ -0,0 +1,112 @@
+***************
+Polar Plot Plot
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
Polar Ice Plot
```

File: docs/Users_Guide/polar_plot.rst [hunk 9]
```diff
@@ -0,0 +1,112 @@
+***************
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
**************
```

File: docs/Users_Guide/polar_plot.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+Polar Plot Plot
+***************
```
Comment (jprestop) [inline docs/Users_Guide/polar_plot.rst]:
```suggestion
**************
```

File: docs/Users_Guide/ens_ss.rst [hunk 11]
```diff
@@ -1,4 +1,4 @@
-**************************
+x**************************
```
Comment (jprestop) [inline docs/Users_Guide/ens_ss.rst]:
```suggestion
**************************
```

**Comment by jprestop** [comment]
@bikegeek I finally made it through testing everything.

There are still several references to $METPLOTPY_SOURCE/METplotpy in  that need to be changed to $METPLOTPY_BASE on these pages:

- https://metplotpy.readthedocs.io/en/feature_172_documentation/Users_Guide/performance_diagram.html
- https://metplotpy.readthedocs.io/en/feature_172_documentation/Users_Guide/reliability_diagram.html
- https://metplotpy.readthedocs.io/en/feature_172_documentation/Users_Guide/roc_diagram.html

The Table of Contents in the feature branch still needs to be fixed. I added a suggestion above that you should just be able to accept to resolve the problem.  There was an "x" in front of the asterisks in the header.

Running the ROC diagram fails for me:
```
Orion-dtn-4[82] jprestop$ python  ${METPLOTPY_SOURCE}/metplotpy/plots/roc_diagram/roc_diagram.py ./custom_roc_diagram.yaml

while parsing a block mapping
  in "./custom_roc_diagram.yaml", line 4, column 1
expected <block end>, but found '<block mapping start>'
  in "./custom_roc_diagram.yaml", line 10, column 2
Traceback (most recent call last):
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation/metplotpy/plots/roc_diagram/roc_diagram.py", line 480, in <module>
    main()
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation/metplotpy/plots/roc_diagram/roc_diagram.py", line 469, in main
    r = ROCDiagram(docs)
UnboundLocalError: local variable 'docs' referenced before assignment
```

The Wind Rose plots, for the minimal_wind_rose.yaml file needs to have the points path added:
```
Orion-dtn-4[153] jprestop$ python $METPLOTPY_BASE/metplotpy/plots/wind_rose/wind_rose.py ./minimal_wind_rose.yaml
Traceback (most recent call last):
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation//metplotpy/plots/wind_rose/wind_rose.py", line 431, in <module>
    main()
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation//metplotpy/plots/wind_rose/wind_rose.py", line 424, in main
    plot.write_output_file()
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation//metplotpy/plots/wind_rose/wind_rose.py", line 365, in write_output_file
    self._save_points(points, filename)
  File "/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation//metplotpy/plots/wind_rose/wind_rose.py", line 386, in _save_points
    with open(output_file, "w+") as f:
OSError: [Errno 30] Read-only file system: '/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation/test/wind_rose/point_stat_mpr.points1'
```
When I added points_path and reran, it worked fine.

I was able to successfully run the Hovmoeller plot with the following command, however it wrote the output to the current directory instead of allowing the user to choose an output directory.  It also wrote to standard output the usage statement for some reason and said ""METPLOTPY_BASE needs to be set to your METplotpy directory" implying that METPLOTPY_BASE was not set to the correct directory even though it is and it gave a deprecation warning.  See below:
```
Orion-dtn-4[113] jprestop$ python $METPLOTPY_BASE/metplotpy/plots/hovmoeller/hovmoeller.py --config ./hovmoeller_defaults.yaml --datadir /work/noaa/ovp/jprestop/METplus/METplus-4.1_sample_data/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller  --input /work/noaa/ovp/jprestop/METplus/METplus-4.1_sample_data/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller/precip.erai.sfc.1p0.2x.2014-2016.nc
usage: hovmoeller.py [-h] [--config CONFIG] [--datadir DATADIR] --input INPUT
                     [--logfile LOGFILE] [--debug]

Hovmoeller diagram

optional arguments:
  -h, --help         show this help message and exit
  --config CONFIG    configuration file
  --datadir DATADIR  top-level data directory (default $DATA_DIR)
  --input INPUT      input file name
  --logfile LOGFILE  log file (default stdout)
  --debug            set logging level to debug

METPLOTPY_BASE needs to be set to your METplotpy directory
INFO:root:/work/noaa/ovp/jprestop/METplus/METplus-4.1_sample_data/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller
INFO:root:/work/noaa/ovp/jprestop/METplus/METplus-4.1_sample_data/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller/precip.erai.sfc.1p0.2x.2014-2016.nc
INFO:root:{'plot_filename': '/work/noaa/ovp/jprestop/tmp/metplotpy/output/erai_precip.png', 'height': 800, 'width': 1200, 'title': 'ERAI Precipitation', 'font_size': 20, 'xaxis_title': 'Longitude', 'yaxis_title': 'Time', 'date_start': datetime.date(2016, 1, 1), 'date_end': datetime.date(2016, 3, 31), 'lat_max': 5, 'lat_min': -5, 'var_name': 'precip', 'var_units': 'mm / day', 'unit_conversion': 250, 'contour_min': 0.2, 'contour_max': 1.6, 'contour_del': 0.2, 'colorscale': 'BuPu'}
INFO:root:Opening /work/noaa/ovp/jprestop/METplus/METplus-4.1_sample_data/model_applications/s2s/UserScript_obsPrecip_obsOnly_Hovmoeller/precip.erai.sfc.1p0.2x.2014-2016.nc
/apps/contrib/MET/METplotpy/METplotpy-feature_172_documentation//metplotpy/plots/hovmoeller/hovmoeller.py:97: DeprecationWarning: parsing timezone aware datetimes is deprecated; this will raise an error in the future
  ts = (time - np.datetime64('1970-01-01T00:00:00Z')) / np.timedelta64(1, 'h')
```

I didn't test the S2S blocking, the weather regime, the spacetime (cross-spectra), the MJO, TC_RMW cross-section, stratosphere diagnostics, or polar ice functionality.

**Comment by bikegeek** [comment]
@jprestop: For the **hovmoeller plot**,  I modified the instructions. The user will explicitly set the full path to the plot_filename setting.  The deprecation warning is from an external Python package.  The other output is informational and I will not remove this because we are in a code-freeze and this requires modifying the hovmoeller.py code (and this is not a bug, just somewhat annoying).  I indicated what these messages are in the documentation (except for the deprecation, as this will resolve when we upgrade to higher versions of all packages with the migration to Python 3.8)

**Pull request merged**